contracts/index_executor/src/lib.rs

```diff
@@ -296,8 +281,59 @@ mod index_executor {
             Ok(())
         }
 
+        #[ink(message)]
+        pub fn pause_executor(&mut self) -> Result<()> {
+            self.ensure_owner()?;
+            self.ensure_running()?;
+            self.is_paused = true;
+            Ok(())
+        }
+
+        #[ink(message)]
+        pub fn unpause_executor(&mut self) -> Result<()> {
+            self.ensure_owner()?;
+            self.ensure_paused()?;
+            self.is_paused = false;
+            Ok(())
+        }
+
+        /// Submit transaction to execute ERC20 approve on behalf of the call on an EVM chain.
+        #[ink(message)]
+        pub fn worker_approve(
```

Comment:
when do we call this method? before a task or within a task?

Reply:
Only for EVM, when we are going to support new bridge or dex, or new asset. The worker account needs to approve those smart contracts. This will be done many times, whenever we want to add more support. And in production, this should be triggered by on-chain governance.